pkg/nsx/services/vpc/vpc.go

```diff
@@ -1033,29 +1033,30 @@ func (s *VPCService) GetDefaultNSXLBSPathByVPC(vpcID string) string {
 }
 
 func (vpcService *VPCService) EdgeClusterEnabled(nc *common.VPCNetworkConfigInfo) bool {
-	var vpcConnectivityProfile *model.VpcConnectivityProfile
-	var getErr error
-	if err := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+	isRetryableError := func(err error) bool {
 		if err == nil {
 			return false
 		}
-		_, errortype := nsxutil.DumpAPIError(err)
-		if errortype != nil && (*errortype == apierrors.ErrorType_SERVICE_UNAVAILABLE || *errortype == apierrors.ErrorType_TIMED_OUT) {
-			return true
-		} else {
-			return false
-		}
-	}, func() error {
+		_, errorType := nsxutil.DumpAPIError(err)
+		return errorType != nil && (*errorType == apierrors.ErrorType_SERVICE_UNAVAILABLE || *errorType == apierrors.ErrorType_TIMED_OUT)
+	}
+
+	var vpcConnectivityProfile *model.VpcConnectivityProfile
+	if err := retry.OnError(retry.DefaultBackoff, isRetryableError, func() error {
+		var getErr error
 		vpcConnectivityProfile, getErr = vpcService.GetVpcConnectivityProfile(nc, nc.VPCConnectivityProfile)
-		return getErr
-	}); err == nil {
-		log.Info("vpc connectivity profile", "service gateway enable", *vpcConnectivityProfile.ServiceGateway.Enable)
-		return vpcService.IsEnableAutoSNAT(vpcConnectivityProfile)
-	} else {
-		log.Error(getErr, "failed to get vpc connectivity profile", "vpc connectivity profile", nc.VPCConnectivityProfile)
+		if getErr != nil {
+			return getErr
+		}
+		log.Info("VPC connectivity profile retrieved", "profile", *vpcConnectivityProfile)
```

Comment:
```suggestion
		log.V(2).Info("VPC connectivity profile retrieved", "profile", *vpcConnectivityProfile)
```
Do you think this change works?

Reply:
log.V(1) is debug level.